stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDocumentProvider.java

```diff
@@ -71,27 +72,34 @@ public void run() {
     }
   };
 
-  public AndroidDocumentProvider(Application application, ThreadBound enforcer) {
+  public AndroidDocumentProvider(
+      Application application,
+      List<DescriptorProvider> descriptorProviders,
+      ThreadBound enforcer) {
     super(enforcer);
 
     mApplication = Util.throwIfNull(application);
     mDocumentRoot = new AndroidDocumentRoot(application);
 
-    mDescriptorMap = new DescriptorMap()
-        .beginInit()
-        .register(Activity.class, new ActivityDescriptor())
-        .register(AndroidDocumentRoot.class, mDocumentRoot)
-        .register(Application.class, new ApplicationDescriptor())
-        .register(Dialog.class, new DialogDescriptor());
+    mDescriptorMap = new DescriptorMap().beginInit();
```

Comment:
you can still use the fluent "builder syntax" here because `beginInit()` returns `DescriptorMap` and `registerDescriptor()` returns `DescriptorRegistrar`.

and then -- IIRC! -- you can change the return type in `DescriptorMap` so that `registerDescriptor` actually returns `DescriptorMap` even though it's implementing an interface method that returns a narrower type. Java is weird like that (if I'm right, and I'm not 100% sure, but I'm pretty sure I've seen this)